Lib/unittest/mock.py

```diff
@@ -2426,7 +2432,10 @@ def _iter_side_effect():
     handle.__iter__.side_effect = _iter_side_effect
 
     def reset_data(*args, **kwargs):
-        _state[0] = _iterate_read_data(read_data)
+        if isinstance(read_data, bytes):
```

Comment:
Might be good to get a function for this. Something like "_read_data_to_stream" to remove duplication.

Reply:
Thanks :)